python/cudf/cudf/core/column/numerical.py

```diff
@@ -583,6 +595,9 @@ def can_cast_safely(self, to_dtype):
 
         # want to cast float to int:
         elif self.dtype.kind == "f" and to_dtype.kind in {"i", "u"}:
+            if libcudf.unary.is_nan(self).any():
+                return False
```

Comment:
Instead of this change could we use something like `_drop_na_rows(drop_nan=False)` at line 606?

Reply:
Looks like with dropping rows is more expensive than doing an `is_nan.any()` check:
```python
In [1]: import cudf
In [2]: import numpy as np

In [3]: def s1(col):
   ...:     return cudf._lib.unary.is_nan(col).any()
   ...: 

In [4]: def s2(col):
   ...:     return col.dropna(drop_nan=False)
   ...: 

In [5]: s = cudf.Series([1, 2, np.nan, None, np.nan, 324, 123, 45] * 10_00_000_00, nan_as_nul
   ...: l=False)

In [6]: %timeit s1(s._column)
43.8 ms ± 4.74 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [7]: %timeit s2(s._column)
219 ms ± 2.66 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

But an easy solution to this perf issue is to introduce `fill_nan` parameter in `NumericalColumn.fillna`, which will let us control filling of `nan` values. Hence since we are already anyway performing  a `self.fillna`, passing `fill_nan` as `False` will make it more efficient and help us achieve `nan` validation needed here.